working/2364 - primary constructors/feature-specification.md

```diff
@@ -361,14 +397,28 @@ constants *(holding parameter default values)*.
 Where no processing is mentioned below, _D2_ is identical to _D_. Changes
 occur as follows:
 
+Assume that `p` is an optional formal parameter in _D_ that does not have a
+declared type, but it does have a default value whose static type in the
+empty context is a type `T` which is not `Null`. In that case `p` is
+considered to have the declared type `T`. When `T` is `Null`, `p` is
+considered to have the declared type `Object?`.
```

Comment:
If there is no default value (including not being optional) and there is no declared type, is the type `dynamic` or `Object?`?

Reply:
Now using `Object?`, with a remark about the rationale.